### ISSUE TITLE
Username is sometimes missing in queue

### ISSUE THREAD

**Issue by beaufour** [jaedb/Iris#436]
We are a few that listen to the same stream from home, and we fairly often end up with a someone logging in to Iris and then requesting a bunch of songs but without a username. That way we don't know who to complain to if we don't like the music :) It would be really nice if it wasn't possible to log in without a username.

**Comment by jaedb** [comment]
Username _is_ compulsory in order for a connection to be established, however to ensure users can jump straight in and enjoy Iris, a username is auto-generated (eg "123456789") on load.

You could always track who that user was by their IP address?

**Comment by beaufour** [comment]
Oh. We quite often end up with requests in the queue with no username (and not coming from the radio :) ). Not sure how that happens then. Maybe the metadata request for the requestor information fails and isn't displayed?

**Comment by jaedb** [comment]
Are you able to attach a screenshot of this when it happens next? It'd also be useful to know the details of your environment (ie OS, browser, Iris version, etc) as well as any logs/errors in the browser (_File > Developer Tools_) or in the Mopidy server.

**Comment by beaufour** [comment]
Will do

**Comment by eskech** [comment]
running version 3.40.0 and it could be a Safari only problem. We only see it for people using Safari (I am part of the @beaufour group). Could be an problem with access to local storage where the username is written. Will try to provide console logs.

**Comment by beaufour** [comment]
It just happened for me. Added songs to the queue using my iPhone, and they are not attributed to my username. Environment: iPhone 11 Pro, iOS 13.1.3, Safari. Iris 3.43.2.

I hooked up Safari to it, and I think the issue is that WebSockets are blocked :( Safari doesn't seem to support auth and websockets? (https://bugs.webkit.org/show_bug.cgi?id=80362) I've put the installation behind `nginx` with auth enabled. So that's probably what goes wrong in this case.
